restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/ua/UserAgentManager.java

```diff
@@ -158,8 +160,12 @@ private void clean() {
                     if (logger.isInfoEnabled()) {
                         logger.info("Registration: " + result.getAddressOfRecord() + " didn't respond to OPTIONS and will be removed now");
                     }
-                    registrations.removeRegistration(result);
-                    monitoringService.tell(new UserRegistration(result.getUserName(), result.getLocation(), false), self());
+                    // Instead of removing registrations we ping the client one last time to ensure it was not a temporary loss
+                    // of connectivity. We don't need to remove the registration here. It will be handled only if the OPTIONS ping
+                    // times out and the related calls from the client cleaned up as well
+                    ping(result.getLocation());
```

Comment:
There was a nasty race condition with the existing code, where the registration would be removed but the call never cleaned up from the DB. Now we make sure to remove the registration only in one place, ie https://github.com/RestComm/Restcomm-Connect/blob/d85949282fadc41807293ef4734cec5f51c0653a/restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/ua/UserAgentManager.java#L275 which also guarantees the active call is cleaned up from memory and DB. 

The reasoning is to make a last try to ping the client before fully giving up.

This has been heavily tested live on conference calls with various participants dropping in conditions like wifi loss, tab closing